scripts/bootstrap/compile.sh

```diff
@@ -254,6 +254,12 @@ EOF
   link_file "${PWD}/src/MODULE.tools" "${BAZEL_TOOLS_REPO}/MODULE.bazel"
   new_hash=$(shasum -a 256 "${BAZEL_TOOLS_REPO}/MODULE.bazel" | awk '{print $1}')
   sed -i.bak "/\"bazel_tools\":/s/\"[a-f0-9]*\"/\"$new_hash\"/" MODULE.bazel.lock
+  # TODO: Temporary hack for lockfile version mismatch, remove these lines after updating to 7.1.0
+  sed -i.bak 's/"lockFileVersion": 3/"lockFileVersion": 4/' MODULE.bazel.lock
+  sed -i.bak 's/^\s*"name": "[^"]*~[^"]*",//' MODULE.bazel.lock
+  # Also remove the name attribute of the repo backing the platforms module, which is "well-known"
+  # and thus doesn't contain a tilde.
+  sed -i.bak 's/          "name": "platforms",//' MODULE.bazel.lock
```

Comment:
This is reaching a level of hackery where I would usually reach for other tools - is something like `jq` available in CI?

Reply:
My sed foo is too weak to come up with an expression that also works on macOS.